CONTRIBUTING.md

```diff
@@ -1,73 +1,70 @@
-git push --force-with-lease origin my-new-feature
-```
+Follow the documentation in the development document for the [pre-requisites](./Development.md#Prerequisites) required for the setup, as well [building](./Development.md#building) the project.
 
-### Updating pull requests
+### Run test
 
-If your PR fails to pass CI or needs changes based on code review, you'll most likely want to squash these changes into
-existing commits.
+[How To run unit and e2e test](./DEVELOPMENT.md#testing)
 
-If your pull request contains a single commit or your changes are related to the most recent commit, you can simply
-amend the commit.
+Before a PR is accepted it will need to pass the all the checks on the CI.
 
-``` shell
-git add .
-git commit --amend
-git push --force-with-lease origin my-new-feature
-```
 
-If you need to squash changes into an earlier commit, you can use:
+## Contribution Flow
 
-``` shell
-git add .
-git commit --fixup <commit>
-git rebase -i --autosquash main
-git push --force-with-lease origin my-new-feature
-```
+This project operates according to the talk, then code rule. If you plan to submit a pull request for anything more than a typo or obvious bug fix, first you should [raise an issue][new-issue] to discuss your proposal, before submitting any code.
+
+This is a rough outline of what a contributor's workflow looks like:
 
-Be sure to add a comment to the PR indicating your new changes are ready to review, as GitHub does not generate a
-notification when you git push.
+* Make a fork of the repository within your GitHub account
+* Create a topic branch in your fork from where you want to base your work
+* Make commits of logical units
+* Make sure your commit messages are with the proper format, quality and descriptiveness (see below)
+* Push your changes to the topic branch in your fork
+* Create a pull request containing that commit
 
-### Code Style
+We follow the GitHub workflow and you can find more details on the [GitHub flow documentation](https://docs.github.com/en/get-started/quickstart/github-flow).
 
-### Formatting Commit Messages
+Before submitting your pull request, we advise you to use the following:
 
-We follow the conventions on [How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/).
+### Pull Request Checklist
 
-Be sure to include any related GitHub issue references in the commit message.  See
-[GFM syntax](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for referencing issues
-and commits.
+1. Check if your code changes will pass both code linting checks and unit tests.
+2. Ensure your commit messages are descriptive. We follow the conventions on [How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/). Be sure to include any related GitHub issue references in the commit message. See [GFM syntax](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for referencing issues and commits.
```

Comment:
I would change this bullet point with out conventional commit standard as follows.

Commit messages should follow the following format:
```
type: <short-summary>
```

Add ! after type if it's a breaking change
```
type!: <short-summary>
```

**NOTE**: `short-summary` should explain what the commit does. The automation tools we build will basically take this `short-summary` and put it in a changelog. So write your `short-summary` in a way that is worthy to go in the changelog.

`type`: can be one of the following:
- `build`: Changes that affect the build system (example: Makefile)
- `chore`: Housekeeping tasks, grunt tasks etc; no production code change
- `ci`: Changes to our CI configuration files and scripts
- `docs`: Documentation only changes
- `deprecate`: A code change that deprecates a feature
- `feat`: A new feature
- `fix`: A bug fix
- `perf`: A code change that improves performance
- `refactor`: A code change that neither fixes a bug nor adds a feature, includes updates to a feature.
- `removed`: Removed a feature that was previously deprecated
- `test`: Adding missing tests or correcting existing tests
```